src/main/scala/viper/gobra/util/ViperChopper.scala

```diff
@@ -0,0 +1,644 @@
+          val pbodies = vertices.collect{ case v: PredicateBody => predicateTable(v.predicateName) }.toSeq
+          val filteredSigs = psigs.filterNot(sig => pbodies.exists(_.name == sig.name))
+          pbodies ++ filteredSigs
+        }
+        val fields = vertices.collect{ case v: Field => fieldTable(v.fieldName) }.toSeq
+        val domains = {
+          val fs = vertices.collect{ case v: DomainFunction => domainFunctionTable(v.funcName) }.toSeq.groupMap(_._2)(_._1)
+          val as = vertices.collect{ case v: DomainAxiom => (v.v, v.d)}.toSeq.groupMap(_._2)(_._1)
+          val ds = vertices.collect{ case v: DomainType => domainTable(v.v.domainName) }.toSeq
+          val totalDs = (ds ++ fs.keys ++ as.keys).distinct
+
+          totalDs.map{ d =>
+            d.copy(functions = fs.getOrElse(d, Seq.empty), axioms = as.getOrElse(d, Seq.empty))(d.pos, d.info, d.errT)
+          }
+        }
+
+        program.copy(
+          methods = methods,
+          functions = funcs,
+          predicates = preds,
+          fields = fields,
+          domains = domains
+        )(program.pos, program.info, program.errT)
+      }
+    }
+  }
+
+  type Edge[T] = (T, T)
+  object Edges {
+
+    /** Returns all dependencies induced by a member. */
+    def dependencies(member: vpr.Member): Seq[Edge[Vertex]] = {
+      member match {
+        case m: vpr.Method =>
+          {
+            val from = Vertex.Method(m.name)
+            usages(m).map(from -> _)
+          } ++ {
+            val from = Vertex.MethodSpec(m.name)
+            (m.pres ++ m.posts ++ m.formalArgs ++ m.formalReturns).flatMap(exp => usages(exp).map(from -> _))
+          }
+
+        case f: vpr.Function =>
+          val from = Vertex.Function(f.name)
+          usages(f).map(from -> _)
+
+        case p: vpr.Predicate =>
+          {
+            val from = Vertex.PredicateSig(p.name)
+            p.formalArgs.flatMap(exp => usages(exp).map(from -> _))
+          } ++ {
+            val from = Vertex.PredicateBody(p.name)
+            usages(p).map(from -> _)
+          } ++ { Seq(Vertex.PredicateBody(p.name) -> Vertex.PredicateSig(p.name)) }
+
+
+        case f: vpr.Field =>
+          val from = Vertex.Field(f.name)
+          usages(f).map(from -> _)
+
+        case d: vpr.Domain =>
+          d.axioms.flatMap { ax =>
+            val mid = Vertex.DomainAxiom(ax, d)
+            val tos = usages(ax.exp)
+            val froms = tos // this is just an approximation for now. This could be improved.
+            val finalFroms = if (froms.nonEmpty) froms else Vector(Vertex.Always)
+            finalFroms.map(_ -> mid) ++ tos.map(mid -> _)
+          } ++ d.functions.flatMap { f =>
+            val from = Vertex.DomainFunction(f.name)
+            usages(f).map(from -> _)
+          }
+
+        case _ => Vector.empty
+      }
+    }
+
+    /** Returns all entities referenced in the subtree of node `n`. */
+    def usages(n: vpr.Node): Seq[Vertex] = {
```

Comment:
same here

Reply:
Same reasoning, I push back most ordering operations until after the conversion to ints.